R/grouped_epi_archive.R

```diff
@@ -247,19 +253,9 @@ epix_slide.grouped_epi_archive <- function(x, f, ..., before, ref_time_values,
   }
 
   # Validate and pre-process `before`:
-  if (missing(before)) {
-    cli_abort("`before` is required (and must be passed by name);
-                if you did not want to apply a sliding window but rather
-                to map `epix_as_of` and `f` across various `ref_time_values`,
-                pass a large `before` value (e.g., if time steps are days,
-                `before=365000`).")
+  if (!(anyInfinite(before) || testInt(before, lower = 0L, null.ok = FALSE, na.ok = FALSE))) {
```

Comment:
```suggestion
  if (!(identical(before, Inf) || test_int(before, lower = 0L))) {
```
- No longer accepts -Inf.
- Rest is stylistic.  Suggested here but not completely sure about omitting the redundant args to `test_int()` since it doesn't follow the defaults of the rest of the package (e.g., `test_integerish()`); guess we could accidentally trick ourselves either way.

Reply:
todo: do / open issue to: allow difftimes directly as `before` arguments if the `version` is of a class compatible with difftimes, and maybe disallow int values if `version` is datetime-like.

suggest: also remove `time_step` from `epi_slide*` methods.  Maybe ask Nat first if he foresees gotchas and whether accepting difftimes is required to make some use cases possible.  If not removing in this PR, please open a separate issue.

todo: remove surviving `@param time_step` from `epix_slide` roxygen.

suggest: check over the vignettes (e.g., {slide,archive,advanced}.Rmd, maybe others) and update them if needed.  I briefly looked and didn't see anything that requires an update, but might be worth a second pass.

Reply:
- Definitely feeling low prio on the difftimes thing. Never worked with that data type.
- Will check about `time_step`. 
- Removed the docs line.
- Repo-wide search for `time_step` brings up nothing, so I think we're good on vignettes.

Reply:
One thing that may add to prio of difftimes is that they seem natural for making time windows when the versions and/or time_values are datetimes, and integers don't seem natural or advisable for that.  But there's probably a lot more to think about with datetimes that we've neglected, and it may take some thinking to establish whether difftimes are the best thing to use for `before` in this case or not.

If you do remove difftime support --- and even if you include it with datetime in mind --- maybe double-check that it's not accidentally accepted by the validation code and added in some strange way to non-datetime stuff.

Reply:
I just think we should stay focused on common use cases before generalizing. I have yet to see a version column that's not Date.

Reply:
time_step in epi_slide seems functional and non-trivial, see `test-epi_slide.R` tests comparing epi_slide and epi_slide_mean. Doesn't really make sense to remove.